Add TIMESTAMPDIFF function to searchkit so you can show time between two dates

mattwire opened the pull request:
Overview
----------------------------------------
For example, for scheduled jobs you can show number of seconds it took to run by using "Time between two dates" for `last_run` and `last_run_end`.

Before
----------------------------------------
Cannot show time between two dates (can already show days using "Days between two dates" but that's not the same).

After
----------------------------------------
Can show time between two dates (eg. seconds).

Technical Details
----------------------------------------
Uses the SQL function `TIMESTAMPDIFF` which takes 3 parameters - the format, eg. SECONDS, MINUTES etc. and then the two date fields.

Comments
----------------------------------------
@colemanw I couldn't work out how to get the first field to work properly as a dropdown selector and have temporarily added a hacky `renderExpression()` to fix the query and make it work.
But I'm assuming there's a better way to define it in `params()` so that param1=dropdown selector for SECONDS,MINUTES etc. and then param2 is the actual field and param3 is the field to compare.

mattwire committed:
Add TIMESTAMPDIFF function to searchkit so you can show time between two dates

francescbassas commented:
@mattwire I tested in a SearchKit and it's working great

![imatge](https://github.com/user-attachments/assets/827c3458-21eb-46cc-93d1-d20a9148cc9c)

colemanw commented:
@mattwire thanks for this. I can probably think of a cleaner solution for that. Meanwhile can you just add a line or two to the sqlFunctionTest to cover this function?

mattwire commented:
@colemanw I added a test. And I added some notes to the code.

Basically the "only" problem now is getting that list of SECONDS, MINUTES etc. to display - it's currently defined as a SqlString which is incorrect. Any ideas how I could make that work?

colemanw commented:
@mattwire yes I think I have a fix for that. Will push it up shortly...

colemanw committed:
Api4 - Fix parsing arguments for TIMESTAMPDIFF function

Also fixes SearchKit UI for TIMESTAMPDIFF and EXTRACT sql functions

colemanw commented:
@mattwire I got it working in the API and then had the joyful experience of debugging the angularJs code to get it working there. In the process I found a bug affecting a similar function (EXTRACT). All fixed now.

totten commented:
civibot, test this please

colemanw commented:
Let's go ahead & merge this into 6.0 because it fixes an unreleased UI regression on the same branch.

mattwire commented:
Thanks @colemanw !
